### PR TITLE
Move avatar clearing in the background

### PR DESCRIPTION
**Pull request by CarlSchwan**
TODO
- [x] testing

### CODE REVIEW

**Comment by ArtificialOwl** [comment]
still, if everyone put small jobs right after upgrade ... :]

**Review by ArtificialOwl — APPROVED** [review]

File: lib/private/Repair.php [hunk 1]
```diff
@@ -200,22 +200,22 @@
...
-			\OC::$server->query(EncryptionMigration::class),
-			\OC::$server->get(ShippedDashboardEnable::class),
-			\OC::$server->get(AddBruteForceCleanupJob::class),
-			\OC::$server->get(AddCheckForUserCertificatesJob::class),
-			\OC::$server->get(RepairDavShares::class),
-			\OC::$server->get(LookupServerSendCheck::class),
-			\OC::$server->get(AddTokenCleanupJob::class),
+			\OCP\Server::get(ResetGeneratedAvatarFlag::class),
+			\OCP\Server::qet(EncryptionLegacyCipher::class),
+			\OCP\Server::qet(EncryptionMigration::class),
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply lib/private/Repair.php:212]: Sight...

**Comment by CarlSchwan** [comment]
> Could be considered to mark it as slow-running job that will only run in the maintenance window, but I think it should not run too long in most production cases either, so would be good with me.

This is not implemented for QueuedJob only TimmedJob

**Comment by PVince81** [comment]
@CarlSchwan can you test and tick the checkbox to confirm ?

**Comment by CarlSchwan** [comment]
> @CarlSchwan can you test and tick the checkbox to confirm ?

tested (by manually increasing the version) and it work, job is added and executed on the next cron run

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by blizzz** [comment]
NoDB tests are failing

**Comment by nickvergessen** [comment]
Manual backport in https://github.com/nextcloud/server/pull/34192

**Comment by nickvergessen** [comment]
Backport was already done in https://github.com/nextcloud/server/pull/34192

**Pull request merged**